File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        long res = 0;
        Stack<Integer> stack = new Stack<>();
        
        for(int i=0; i<N; i++) {
            int h = sc.nextInt();
            while(!stack.isEmpty() && stack.peek() <= h) stack.pop();
            stack.push(h);
            res += stack.size()-1;
        }

        System.out.println(res);
    }
}

File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.swift
let N = Int(readLine()!)!
var stack:[Int] = []
var res = 0

for _ in 0..<N {
    let h = Int(readLine()!)!
    while(!stack.isEmpty && stack[stack.count-1] <= h) { stack.removeLast() }
    stack.append(h)
    res += stack.count - 1
}
print(res)